src/apps/external_executor/protobuf/status.proto

```diff
@@ -2,7 +2,7 @@ syntax = "proto3";
 
 import "google/protobuf/any.proto";
 
-package ccf;
+package ccf.protobuf;
```

Comment:
OK, I see your point for the `Status` as this will be baked in `libccf` eventually. However, I don't think it should be the case for the external executor service as it makes the REST paths more clunky. Separating the gRPC service definition from the protobuf types seems like the way to go to avoid this?

Reply:
> However, I don't think it should be the case for the external executor service as it makes the REST paths more clunky. 

I think this is a fair point that I didn't think of before. I do think we should try to make it much less verbose. I'm not yet sure of what that would be though.

Reply:
I would think `package ccf.externalexecutor` for the service definition (or the new name if it's renamed) and `package ccf.externalexecutor.protobuf` for the data types?

Reply:
I also think the clunky URLs aren't ideal, but its acceptable since they're hidden by a client library? So its annoying that we have to use the URL `/externalexecutor.protobuf.ExecutorRegistration/RegisterExecutor`, but actually nobody uses that directly - they call `ExecutorRegistration.RegisterExecutor()` in their client? We could split the files and have multiple packages, but is `/externalexecutor.ExecutorRegistration/RegisterExecutor` so much better that it's worth the split?

Reply:
My understanding is that these `/externalexecutor.protobuf.ExecutorRegistration/RegisterExecutor` urls are exposed publicly, so they _can_ be accessed externally? If that is never going to happen, then this PR as it stands looks good.

Reply:
They are public and external, but not used directly by (/visible to) most users. They are public endpoints, and people _could_ call them by hand if they really wanted to, but we assume everyone can and will use a wrapper library which hides the ugly URL.

(Unrelated addendum: We should hide all of these gRPC endpoints from the generated OpenAPI!)